### DISCUSSION TITLE
Allow specifying priorityClassName for Cluster pods

### DISCUSSION THREAD

**Discussion by itay-grudev** [cloudnative-pg/cloudnative-pg#1609] [Ideas]
While we are able to specify `resources` for a Cluster which can grant pods a `Guaranteed` QoS status, during node draining or node downtime, there may be high resource contention and database pods may not be scheduled because the default `priorityClassName` is likely lower than the user would set for a database their app depends on.

A `priorityClassName` field added to the `Cluster` resource would make a lot of sense to address the issue.

Is there a reason or a technical limitation why it wasn't added in the first place? I was looking in the documentation or GitHub, but couldn't find anything.

**Comment by sxd** [comment]
@itay-grudev hi!

Can you give us a clear example of what are you proposing? Because this sounds a lot like a PriorityClass
I'll turn this into a discussion so we can create a proper issue with the right definitions of what it's requested

Regards!

**Reply by jakubhajek** [reply]
I also vote to have a possibility to configure a priorityClassName for cluster-related resources. We can configure it for [the operator ](https://github.com/cloudnative-pg/charts/commit/8b6009a60f687097b7eee84f24ca8dfa14479a1b) but that feature is missing for clusters. Thanks in advance for creating a feature request to add that enhancement.

**Reply by itay-grudev** [reply]
@sxd Yes, I am referring to a `PriorityClass` which seems to be missing as an option for Clusters.

**Reply by itay-grudev** [reply]
Is there a practical concern against adding a `PriorityClass` option?

**Reply by batistein** [reply]
We would also need this to ensure that databases have higher priority than other applications. This is really necessary if you operate larger cluster...

**Reply by tonygilkerson** [reply]
For the project I am working on priorityClassName is required.  Why was this converted to a discussion? I feel this is an issue.

**Reply by Simonzhaohui** [reply]
We hope our postgresql cluster is configured with high priority and  Guaranteed QoS using `priorityClassName`

**Reply by jakubhajek** [reply]
There are two factors discussed in this topic, [QOS](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/) and [PrioteryClasses](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass). Basically speaking  QOS is related to the resource constraints such as memory and CPU vs PriorityClass which indicates the importance of a pod in relative to the other pods. 
We can configure [resources](https://cloudnative-pg.io/documentation/1.20/resource_management/)  in the cluster resource by setting memory and CPU limits equal to the memory and CPU requests and having Guaranteed QOS.

By opening that discussion we would like to have the possibility to configure PriorityClasses for the CNPG custom resource such as cluster and pooler since they are crucial components. A similar feature has been added to the operator itself in this [PR](https://github.com/cloudnative-pg/charts/pull/65).

**Reply by litaocdl** [reply]
IIUC with prioryClass we can then provide the possibility that postgres pod would be the latest one been evicted or if there is a resource crises, other lower priority pod will be evicted but make sure posgres pod could be scheduled and run

**Comment by phisco** [comment]
FYI there is a PR in flight to address this: https://github.com/cloudnative-pg/cloudnative-pg/pull/2043

**Comment by itay-grudev** [comment]
@phisco I'm closing this as completed. Thanks!
